### PR TITLE
Add reader sketch

### PR DESCRIPTION
**Pull request by cyx**
Just an idea, there is a way to avoid channels and the like, but this introduces the least amount of changes while still riding the same broker implementation.

### CODE REVIEW

File: broker/reader.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+func (r *Reader) Read(p []byte) (n int, err error) {
+	msg, ok := <-r.ch
+
+	if !ok {
+		err = io.EOF
+	}
+
+	if n = len(msg); n > 0 {
```
Comment (tt) [inline broker/reader.go]: What if the message is larger than `p`?

File: broker/reader.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package broker
+
+import (
+	"io"
+
+	"github.com/heroku/busl/util"
+)
+
+type Reader struct {
```
Comment (tt) [inline broker/reader.go]: This can be unexported.

**Comment by tt** [comment]
This does give convenience if you, say, want to use `io.Copy`, but I still wonder how applicable this is to our known use cases. For the chunked encoding, we could do just copy, but then we'd miss the heartbeats. For Server-Sent Events, I think pushes are definitely better than polling. It seems we'd end up having a lot more code without any cleaner usage. Perhaps you can spike the intended usage to demonstrate that it would be cleaner?

**Comment by cyx** [comment]
Yes I can add to this spike a short intended usage.

For the heartbeats was thinking of creating a KeepAliveReader of sort to abstract that part, which can accept the null byte (and we provide :keepalive in the case of an event-stream request).

**Comment by cyx** [comment]
I'll leave this open for now but I'm tempted to just close it :-)

**Pull request closed without merging**